### DISCUSSION TITLE
Hiwonder TTL Board USB voltage

### DISCUSSION THREAD

**Discussion by stratomancer** [Roger-random/Sawppy_Rover#21] [Q&A] [answered]
Hello,

I am connecting up my Hiwonder TTL board and planning to use the USB interface. The configuration is the same as the Raspberry Pi example in the SAWPPY main docs. Unfortunately I have noticed with my multimeter that I am getting the same ~7.4 volts from the battery power source on the USB's 5v line. This is causing problems for the RPi since this voltage is way over USB spec. Has anyone seen this voltage before? I would expect the TTL board to keep the two voltages separate.

**Comment by srmainwaring** [comment, marked as answer]
I’ve seen this issue before when the boards fail. The bus linker regulator circuit can fail and sends the power voltage to the UART, it cost me a flight controller in the past. I now use a bidirectional logic level shifter board to isolate the bus linker UART from the microcontroller (bus linker on the high voltage side). Opto isolation would be better still.

**Reply by stratomancer** [reply]
Thought I saw magic smoke last night. Your response helps. Thanks, @srmainwaring. Do you have a bi-dir logic level shifter you recommend and are you controlling with micro controller (e.g. Arduino) on the TTL Debugger's UART?

**Reply by srmainwaring** [reply]
I used this one from sparkfun as I had a couple from other projects: https://www.sparkfun.com/products/12009

Usually they’re for connecting a RPi UART to an Arduino but here I’m using it as a failsafe to stop the MCU from getting fried. Unfortunately if the bus linker is reading 7.4V on the UART it’s gone.

I have used an Arduino to interface to the buslinker. I found the USB port on debug board to be too slow for good control. I have a ROS stack that I’ve been using but have been recently experimenting with using ArduPilot to control the rover with Mateksys H743-WING FCU.

**Reply by stratomancer** [reply]
Nice. I'm working on ROS2 stack for my rover. Running crash project mode to try to have my rover working by Perseverance landing on Mars.  I've seen a few people posting about the ArduPilot and that looks quite interesting.

**Reply by Roger-random** [reply]
Yikes! That is a horrible failure mode.

I haven't experienced this myself and I'm sorry you guys lost hardware due to it. But I'm thankful for the knowledge.

I see this as yet another reason to move away from using LewanSoul/Hiwonder for future Sawppy evolution.

**Reply by stratomancer** [reply]
I my case, last night I was wiring in a INA3221 upstream on my power distribution bus for Voltage & Current monitoring telemetry. I got something wrong with the ground (I think, still not sure) on the INA3221 channel connected to the TTL board. I saw a brief little wisp of smoke on the TTL before I snatched the USB cable out of the computer. Decided it was time stop for the evening and started checking with multimeter this morning and noticed the volts bleeding over onto the USB.

**Reply by stratomancer** [reply]
Aside from the proprietary protocol and control board, I like the features and easy wiring of the LewanSoul/Hiwonder servos. Wish they used more open I2C protocol for the servos, though.
